iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -490,17 +491,30 @@ public List<VirtualColumn> acidVirtualColumns() {
   }
 
   @Override
-  public List<FieldSchema> acidSelectColumns(org.apache.hadoop.hive.ql.metadata.Table table) {
-    // TODO: make it configurable whether we want to include the table columns in the select query
-    // it might make delete writes faster if we don't have to write out the row object
-    return Stream.of(ACID_VIRTUAL_COLS_AS_FIELD_SCHEMA, table.getCols())
-        .flatMap(Collection::stream)
-        .collect(Collectors.toList());
+  public List<FieldSchema> acidSelectColumns(org.apache.hadoop.hive.ql.metadata.Table table, Operation operation) {
```

Comment:
nit: we could rename this, the other functions and class attributes using the `acid` keyword to something more general, like `virtualMetadataSelectColumns`. We can do this in a separate PR.